File: gsw/GENERIC_IMU/procedures/generic_imu_test.rb
require 'cosmos'
require 'cosmos/script'
require 'mission_lib.rb'

class IMU_LPT < Cosmos::Test
  def setup
    
  end

  def test_lpt
    start("tests/generic_imu_lpt_test.rb")
  end

  def teardown

  end
end

class IMU_CPT < Cosmos::Test
  def setup
      
  end

  def test_cpt
    start("tests/generic_imu_cpt_test.rb")
  end

  def teardown

  end
end

class Generic_imu_Test < Cosmos::TestSuite
  def initialize
      super()
      add_test('IMU_CPT')
      add_test('IMU_LPT')
  end

  def setup
  end
  
  def teardown
  end
end

File: gsw/GENERIC_IMU/procedures/generic_imu_test.py
import sys
import glob

for p in glob.glob('/gems/gems/openc3-cosmos-nos3-*/targets/GENERIC_IMU/scripts'):
    if p not in sys.path:
        sys.path.append(p)

from openc3.script.suite import Suite, Group

try: 
    from nos3.generic_imu_lib import safe_generic_imu
    from nos3.generic_imu_app_test import run_generic_imu_app_test
    from nos3.generic_imu_device_test import run_generic_imu_device_test
    from nos3.generic_imu_ast_test import run_generic_imu_ast_test
except ImportError:
    pass

class GENERIC_IMU_Functional_Test(Group):
    def setup(self):
        safe_generic_imu()

    def script_application(self):
        run_generic_imu_app_test()

    def script_device(self):
        run_generic_imu_device_test()

    def teardown(self):
        safe_generic_imu()

class GENERIC_IMU_Automated_Scenario_Test(Group):
    def setup(self): 
        safe_generic_imu()

    def script_ast(self):
        run_generic_imu_ast_test()

    def teardown(self):
        safe_generic_imu()

class Generic_imu_Test(Suite):
    def __init__(self):
        super().__init__()
        self.add_group(GENERIC_IMU_Functional_Test)
        self.add_group(GENERIC_IMU_Automated_Scenario_Test)

    def setup(self):
        safe_generic_imu()
  
    def teardown(self):
        safe_generic_imu()